src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -49,16 +49,17 @@ Copyright (c) .NET Foundation. All rights reserved.
   </PropertyGroup>
 
   <!-- For .NET 8 and higher, we will by default use a simplified "portable" RID graph -->
-  <PropertyGroup Condition="'$(UseRidGraph)' == ''">
-    <UseRidGraph Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '8.0'))">false</UseRidGraph>
-    <UseRidGraph Condition="'$(UseRidGraph)' == ''">true</UseRidGraph>
+  <PropertyGroup>
+    <_UseRidGraph>$(UseRidGraph)</_UseRidGraph>
+    <_UseRidGraph Condition="'$(_UseRidGraph)' == '' AND '$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '8.0'))">false</_UseRidGraph>
+    <_UseRidGraph Condition="'$(_UseRidGraph)' == ''">true</_UseRidGraph>
```

Comment:
Rather than introducing a separate `_UseRidGraph` property, I think it would be clearer to add a `UseRidGraphWasSpecified` property that is set to true if `UseRidGraph` is not empty before the defaulting logic kicks in.